server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/migration/MappingsSourcesMigration.java

```diff
@@ -58,7 +86,28 @@ public Result run() {
     private Mono<Result> migrate(Pair<MappingSource, Mapping> mappingEntry) {
         return cassandraMappingsSourcesDAO.addMapping(mappingEntry.getRight(), mappingEntry.getLeft())
             .map(any -> Result.COMPLETED)
-            .doOnError(e -> LOGGER.error("Error while performing migration of mappings sources", e))
+            .doOnSuccess(success -> successfulMappingsCount.incrementAndGet())
+            .doOnError(e -> {
```

Comment:
I suspect do* to be asynchronous, you should rather use synchronous transformations.

Reply:
For logging and progress reporting, being asynchronous does not seem to be an issue, right?

Reply:
it's updated a counter and there's nothing that tells you the counter are eventually updated, leading to future problem for the API consumers

Reply:
I don't really think there is an issue either but why not...

Then well for the error counter, just need to move it to `onErrorResume`. Ok.
However for the success counter, I would have thought that moving the operation to the `map` should be enough (`map` is a synchronous transformation), but then somehow all my tests where the success counter is supposed to be > 0 are failing (like it's never incremented?). I'm not sure to get it myself and couldn't find a better way than `doOnSuccess` so far. So if you have anything better, I'm opened :)

Reply:
You can do that in a map step, extracted in a `Result onSuccess(Result result)` method?

Reply:
As said above, already tested and... it was a fail

Reply:
> I would have thought that moving the operation to the `map` should be enough (`map` is a synchronous transformation)

how about `.flatMap(any -> Mono.fromRunnable(() -> increaseCount)) or `.then(Mono.fromRunnable(() -> increaseCount))`. Both respect synchronization

Reply:
`map` and `flatMap` are indeed not working in this case as previous operation returns a `Mono<Void>`.
I used a `then(Mono.fromCallable())` at the end, to increment the counter and return `Result.COMPLETED` in the same time.